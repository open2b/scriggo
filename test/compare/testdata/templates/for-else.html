{# render #}

{% var xs = []int{10, 20, 30} %}
{% for x in xs %}{{ x }} {% else %}NOT EXPECTED (1){% end for %}

{% xs = []int{} %}
{% for x in xs %}NOT EXPECTED (2){% else %}expected{% end for %}

{%%
    var s = []int{40, 50, 60}
    var f = func() []int {
        toreturn := s
        if len(s) > 0 {
            s = nil
        }
        return toreturn
    }
%%}

{% for _, v := range f() %}
    {{ v }}
{% else %}
    i'm the else block
{% end for %}

{% for _, v := range f() %}
    {{ v }}
{% else %}
    i'm the else block
{% end for %}