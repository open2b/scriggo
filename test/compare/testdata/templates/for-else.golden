10 20 30 

expected


    40
    50
    60

    i'm the else block